src/core/node/nodeRecord.ts

```diff
@@ -88,6 +93,20 @@ export class NodeRecord {
     return this.values.length > 0;
   }
 
+  /**
+   * Update node record.
+   */
+  update(): void {
```

Comment:
As this seems to update only the GroupID and nothing else, how about:
```suggestion
  updateGroupID(): void {
```